spec/rubocop/cop/style/to_json_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::ToJson, :config do
+  it 'registers an offense when using JSON.generate' do
+    expect_offense(<<~RUBY)
+      JSON.generate(foo)
+      ^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo.to_json
+    RUBY
+  end
+
+  it 'recognizes fully qualified reference to JSON' do
+    expect_offense(<<~RUBY)
+      ::JSON.generate(foo)
+      ^^^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+  end
+
+  it 'correctly autocorrects calls with implicit hashes' do
+    expect_offense(<<~RUBY)
+      JSON.generate(hello: :world)
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      { hello: :world }.to_json
+    RUBY
+  end
+
+  it 'correctly autocorrects calls with serialization options' do
+    expect_offense(<<~RUBY)
+      JSON.generate(foo, space_before: '  ', space_after: ' ')
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo.to_json(space_before: '  ', space_after: ' ')
+    RUBY
+  end
+
+  it 'correctly autocorrects calls with serialization options as variable' do
+    expect_offense(<<~RUBY)
+      JSON.generate(foo, opts)
+      ^^^^^^^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo.to_json(opts)
+    RUBY
+  end
+
+  it 'ignores calls to JSON.generate with unexpected arguments' do
```

Comment:
Unlikely edge case, but should we check passing a block?

```ruby
JSON.generate(foo) { nil }
```

Reply:
What behavior would we expect? I think I'm OK with either the cop ignoring this other transforming to `foo.to_json { nil }`.

Reply:
On the other hand, if that code is valid, it probably means that someone has overridden `JSON.generate`, in which case we shouldn't transform it...?

Not sure it's worth investing time in this edge case, but I will if you think it is.

Reply:
> What behavior would we expect?

Probably just ignoring it, but yeah, it's a crazy edge case, so it's probably fine to consider out of scope.